### PR TITLE
LPS-190904 Init editingLanguageId with form defaultLanguageId

### PR DESCRIPTION
**Pull request by georgel-pop-lr**
LPS-190904 Init editingLanguageId with form defaultLanguageId

Related issue: https://liferay.atlassian.net/browse/LPS-190904

### CODE REVIEW

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by georgel-pop-lr** [comment]
Hi @marcelabc,

I see on my local that many tests fails because reducers are undefined:

` TypeError: Cannot read properties of undefined (reading 'reduce')

       6 | export function createReducer(reducers, config) {
       7 | 	return (state, action) =>
    >  8 | 		reducers.reduce((prevState, reducer) => {
         | 		         ^
       9 | 			const nextProperties = reducer(prevState, action, config);
      10 | 
      11 | 			if (prevState === nextProperties) {

      at ../../data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/reducers/createReducer.es.js:8:12
      at updateReducer (../../../node_modules/react-dom/cjs/react-dom.development.js:16591:23)
      at Object.useReducer (../../../node_modules/react-dom/cjs/react-dom.development.js:17346:16)
      at useReducer (../../../node_modules/react/cjs/react.development.js:1623:21)
      at FormProvider (../../data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/core/hooks/useForm.es.js:125:4)
      at renderWithHooks (../../../node_modules/react-dom/cjs/react-dom.development.js:16260:18)
      at updateFunctionComponent (../../../node_modules/react-dom/cjs/react-dom.development.js:18347:20)
      at beginWork$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:20176:16)
      at HTMLUnknownElement.callCallback (../../../node_modules/react-dom/cjs/react-dom.development.js:336:14)
      at innerInvokeEventListeners (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:315:27)
      at invokeEventListeners (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:266:3)
      at HTMLUnknownElementImpl._dispatch (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:212:11)
      at HTMLUnknownElementImpl.dispatchEvent (../../../node_modules/jsdom/lib/jsdom/living/events/EventTarget-impl.js:87:17)
      at HTMLUnknownElement.dispatchEvent (../../../node_modules/jsdom/lib/jsdom/living/generated/EventTarget.js:144:23)
      at Object.invokeGuardedCallbackDev (../../../node_modules/react-dom/cjs/react-dom.development.js:385:16)
      at invokeGuardedCallback (../../../node_modules/react-dom/cjs/react-dom.development.js:440:31)
      at beginWork$$1 (../../../node_modules/react-dom/cjs/react-dom.development.js:25780:7)
      at performUnitOfWork (../../../node_modules/react-dom/cjs/react-dom.development.js:24698:12)
      at workLoopSync (../../../node_modules/react-dom/cjs/react-dom.development.js:24671:22)
      at performSyncWorkOnRoot (../../../node_modules/react-dom/cjs/react-dom.development.js:24270:11)
      at ../../../node_modules/react-dom/cjs/react-dom.development.js:12199:24
      at unstable_runWithPriority (../../../node_modules/scheduler/cjs/scheduler.development.js:697:12)
      at runWithPriority$2 (../../../node_modules/react-dom/cjs/react-dom.development.js:12149:10)
      at flushSyncCallbackQueueImpl (../../../node_modules/react-dom/cjs/react-dom.development.js:12194:7)
      at flushSyncCallbackQueue (../../../node_modules/react-dom/cjs/react-dom.development.js:12182:3)
      at flushPassiveEffectsImpl (../../../node_modules/react-dom/cjs/react-dom.development.js:25420:3)
      at unstable_runWithPriority (../../../node_modules/scheduler/cjs/scheduler.development.js:697:12)
      at runWithPriority$2 (../../../node_modules/react-dom/cjs/react-dom.development.js:12149:10)
      at flushPassiveEffects (../../../node_modules/react-dom/cjs/react-dom.development.js:25361:12)
      at Object.<anonymous>.flushWork (../../../node_modules/react-dom/cjs/react-dom-test-utils.development.js:1041:10)
      at act (../../../node_modules/react-dom/cjs/react-dom-test-utils.development.js:1155:9)
      at render (../../../node_modules/@testing-library/react/dist/pure.js:97:26)
      at Object.<anonymous> (test/js/FieldBase/ReactFieldBase.es.js:247:24)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)`

But not sure how to provide this to FieldBaseWithProvider, could someone provide me some help with this?

Regards.

**Comment by alinedoleron** [comment]
Hello, @georgel-pop-lr.

I executed the steps described on the [LPP](https://liferay.atlassian.net/browse/LPP-50141) but I can still reproduce them. I'm not sure if the PR fixes another specific steps. Please, could you confirm this information? 
For me, the code looks good.
Regarding the broken javascript test, I found a solution, despite of the fix will be in files you have not changes 😁 

on _**modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/Validation.js**_:
change the line 28 to:
`<FormProvider initialState={{formBuilder, validations}} **reducers={[]}**>`

do the same on line 34 of **_[bug_paragraph.webm](https://github.com/liferay-forms/liferay-portal/assets/26491796/fd6d87f4-6f67-4734-9828-e59c54be6570)_**

`<FormProvider initialState={{validations}} reducers={[]}>`

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Hi @georgel-pop-lr. I've made some comments. Thanks!

**Comment by georgel-pop-lr** [comment]
ci:test:sf

**Comment by georgel-pop-lr** [comment]
> Hello, @georgel-pop-lr.
> 
> I executed the steps described on the [LPP](https://liferay.atlassian.net/browse/LPP-50141) but I can still reproduce them. I'm not sure if the PR fixes another specific steps. Please, could you confirm this information? For me, the code looks good. Regarding the broken javascript test, I found a solution, despite of the fix will be in files you have not changed😁
> 
> on _**modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/Validation.js**_: change the line 28 to: `<FormProvider initialState={{formBuilder, validations}} reducers={[]}>`
> 
> do the same on line 34 of **_modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/test/js/Validation/ValidationTextAndNumeric.js_**
> 
> `<FormProvider initialState={{validations}} reducers={[]}>`
> 
>  bug_paragraph.webm

Hi @alinedoleron,

Thanks for checking this pull and helping me with the tests, I pushed the changes you suggested and now they are all green.

Regarding your video, that case will not work as you need to provide a value for the default language so it wasn't supposed to be fixed, and I forgot to remove it from the original cases. You could see more about this on https://liferay.atlassian.net/browse/PTR-4105?focusedCommentId=2178895

I updated the cases on LPS-190904 as there were none of them.

I hope it helps.

Regards.

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Comment by georgel-pop-lr** [comment]
ci:test:forms

**Comment by alinedoleron** [comment]
Hi @georgel-pop-lr! The **failures on poshi** tests seem to be related to the PR. I executed one of them and I realized that when create a form, add a title to it, a then we add a new field, the Forms name is erased. Please could you check this scenario and fix it.

Steps: 

1. Create a new Form
2. Give the form a title and a description: Ex: Form 1 / Description of Form 1
3. Add a field to the form
Actual Result: The form title and description are erased.

Thanks! :blush:

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Hi @georgel-pop-lr. Unfortunately the failures of relevant and forms ci routines are related.  :sweat:
Please, you can check my last comment where I describe it better.
Thanks!

**Comment by alinedoleron** [comment]
ci:test:sf

**Comment by alinedoleron** [comment]
ci:test:relevant

**Comment by alinedoleron** [comment]
ci:test:forms

**Comment by georgel-pop-lr** [comment]
ci:stop

**Comment by georgel-pop-lr** [comment]
> Hi @georgel-pop-lr. Unfortunately the failures of relevant and forms ci routines are related. 😓 Please, you can check my last comment where I describe it better. Thanks!

Hi @alinedoleron,

I used ci:stop as I'm still working on a solution.

I found what is happening but still don't have a final working solution.

Once is ready, I will let you know.

Regards.

**Comment by alinedoleron** [comment]
> > Hi @georgel-pop-lr. Unfortunately the failures of relevant and forms ci routines are related. 😓 Please, you can check my last comment where I describe it better. Thanks!
> 
> Hi @alinedoleron,
> 
> I used ci:stop as I'm still working on a solution.
> 
> I found what is happening but still don't have a final working solution.
> 
> Once is ready, I will let you know.
> 
> Regards.

Oh ok, @georgel-pop-lr ! Thanks for the update 😊. 
Since you're still working I'll just ask for one simple adjustment of typing.
Thanks!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Checkbox/Checkbox.tsx [hunk 1]
```diff
@@ -144,6 +146,7 @@ const Main: React.FC<IProps> = ({
 };
 
 interface IProps extends ICheckboxProps {
+	defaultLanguageId?: any;
```
Comment (alinedoleron) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Checkbox/Checkbox.tsx]: Hi @georgel-pop-lr, we avoid using `any` as a type. Please, replace this line with:

```suggestion
	defaultLanguageId?: Liferay.Language.Locale;
```

Thanks! :blush:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.d.ts [hunk 2]
```diff
@@ -10,6 +10,7 @@ export function FieldBase(props: IFieldBase): JSX.Element;
 interface IFieldBase {
 	accessible?: boolean;
 	children?: any;
+	defaultLanguageId?: any;
```
Comment (alinedoleron) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.d.ts]: Hi @georgel-pop-lr, we avoid using any as a type. Please, replace this line with:
```suggestion
	defaultLanguageId?: Liferay.Language.Locale;
```

**Review by alinedoleron — CHANGES_REQUESTED** [review]

**Comment by georgel-pop-lr** [comment]
ci:test:sf

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.d.ts (hunk 2 above)
Comment (georgel-pop-lr) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.d.ts]: This is now outdates as I'm not changing any more all this files, the new solution is more simple as I changed all the logic on FormInfo component.

Comment (alinedoleron) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.d.ts]: Great, @georgel-pop-lr! :smiley:

**Comment by georgel-pop-lr** [comment]
ci:test:sf

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Comment by georgel-pop-lr** [comment]
ci:test:forms

**Comment by georgel-pop-lr** [comment]
> 

Yes, I just pushed a new solution and I think this should work fine :).

Regards.

**Comment by alinedoleron** [comment]
Hi @georgel-pop-lr, sorry for bothering you again :disappointed_relieved:, but the relevant test routine identified some broken tests. Please could you check this scenario? It seems the one that I'd mentioned before.

Steps:

1. Create a new Form
2. Give the form a title and a description: Ex: Form 1 / Description of Form 1
3. Add a field to the form

Actual Result: The form title and description are being erased. ❌
Expected Result: The form title and description should be displayed.✔️

This is the screenshot of one  of the relevant test result.
image

Thanks!

**Review by alinedoleron — CHANGES_REQUESTED** [review]
Some tests has broken =/

**Comment by georgel-pop-lr** [comment]
ci:test:sf

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Comment by georgel-pop-lr** [comment]
> Hi @georgel-pop-lr. Unfortunately the failures of relevant and forms ci routines are related. 😓 Please, you can check my last comment where I describe it better. Thanks!

Hi  @alinedoleron,

I hope now it is solved, the problem was to have null as init value, so I removed it.

Regards.

**Comment by georgel-pop-lr** [comment]
ci:test:stable

**Comment by georgel-pop-lr** [comment]
ci:test:relevant

**Pull request closed without merging**